python/pyspark/sql/connect/session.py

```diff
@@ -235,6 +235,9 @@ def createDataFrame(
             # If no schema supplied by user then get the names of columns only
             if schema is None:
                 _cols = [str(x) if not isinstance(x, str) else x for x in data.columns]
+            elif isinstance(schema, (list, tuple)) and _num_cols < len(data.columns):
+                _cols = _cols + [f"_{i + 1}" for i in range(_num_cols, len(data.columns))]
```

Comment:
In fact, I guess probably we can do a bit more: need to make sure the user provided column name is not the same as the auto-generated one.

Though the probability of the collision is small.